### DISCUSSION TITLE
How to access this.$q quasar object in setup? - Vue 3 Composition API

### DISCUSSION THREAD

**Discussion by robz95** [quasarframework/quasar#8930] [General - Components / Directives / etc] [answered]
I have a simple component script written in Composition Api, butI do not know, how to access this.$q variable in Composition API. How do I make it `work?

`<script lang="ts">
import { computed, defineComponent, ref, ComputedRef } from 'vue';

export default defineComponent({
  name: 'QuasarTest',
  setup() {
    const isMobile: ComputedRef<boolean> = computed((): boolean => {
      return this.$q.screen.xs || this.$q.screen.sm;;
    });
    return {
      isMobile,
      model: ref(null),
    };
  }
});
</script>`

**Comment by metalsadman** [comment, marked as answer]
use the composable [useQuasar](https://next.quasar.dev/vue-composables/use-quasar), sample [usage](https://next.quasar.dev/options/screen-plugin#usage).

**Reply by robz95** [reply]
Thanks a lot

**Reply by devJGuerrero** [reply]
Hello, a question I am following the example that you expose in the comment but I have an error when I call import { useQuasar } from "quasar" does not recognize the object 
![image](https://user-images.githubusercontent.com/29376716/122146091-08faea80-ce1c-11eb-877d-c01353a8089b.png)
![image](https://user-images.githubusercontent.com/29376716/122146164-28921300-ce1c-11eb-9750-1fac172305d6.png)
![image](https://user-images.githubusercontent.com/29376716/122146190-33e53e80-ce1c-11eb-9de5-c76f30634a78.png)
![image](https://user-images.githubusercontent.com/29376716/122146222-42335a80-ce1c-11eb-9628-b058a7f4c127.png)
I am learning about the frame, I appreciate your kind collaboration, Happy evening.

**Reply by metalsadman** [reply]
@devJGuerrero  this is only for Quasar V2(Vue 3).

**Reply by devJGuerrero** [reply]
I thought I had version 2, I followed all the installation steps of quasar documentation 2 https://next.quasar.dev/start/quasar-cli

But thanks for the very kind help.

![image](https://user-images.githubusercontent.com/29376716/122150895-30ee4c00-ce24-11eb-80b7-3fe4bb0cb2ca.png)

**Reply by devJGuerrero** [reply]
My mistake was in not reading this part correctly

![image](https://user-images.githubusercontent.com/29376716/122151096-89254e00-ce24-11eb-9d42-b7031e38d77e.png)
